src/components/swipe.event.jsx

```diff
@@ -0,0 +1,109 @@
+import { Component } from 'react'
+
+class SwipeEventEmitter extends Component {
```

Comment:
I can see that you've tried to use hammer, what problems did you face with SSR?

Reply:
the one that Ryan Dahl regrets about :smile: window is undefined

Reply:
you can check if you are in the browser if call it's functions

Reply:
we have package `can-use-dom` for that

Reply:
hmm, for Hammer I dunno, it may help, but for other libraries that I've tried, they try to use window in initialization time, and on demand. I'll try it with Hammer

Reply:
shit, hammer also failed buil even if I just import it without even using it

Reply:
You see, I don't like my implementations too, especially that this is limited only too horizontal movements, but maybe we may leave it for now, I'm going to continue looking for a good ssr library, because we still need a good lib for photo carousel and other animations.